### DISCUSSION TITLE
Chaining scopes with OR

### DISCUSSION THREAD

**Discussion by rida-elbahtouri** [jeremyevans/sequel#2091] [General]
## Description

when we have scopes like this
```ruby
  def unassigned
      where(assigned_at: Time.zone.now..12.hours.ago).or(assigned_at: nil)
   end
```

Only the query before `or` will be applied to existing data, and the query after OR seems to be applied on all records, in the above case it will be the equivalent of this:
```ruby
 existing_query.where(assigned_at: Time.zone.now..12.hours.ago) + Case.where(assigned_at: nil) 
   
```

I updated that to this:
```ruby
  def unassigned
      case_ids = where(assigned_at: Time.zone.now..12.hours.ago).or(assigned_at: nil).select_map(:id)
      where(id: case_ids)
   end
```

this fixed the issue, but I am wondering if there is a better way to handle this.

**Comment by janko** [comment]
The `#or` here will be applied at level of the whole dataset, it won't regard the last `#where` filter. What you want is build an `OR` expression that you pass to `#where`:

```rb
where(Sequel[assigned_at: Time.zone.now..12.hours.ago] | Sequel[assigned_at: nil])
```

See https://sequel.jeremyevans.net/rdoc/files/doc/sql_rdoc.html#label-Boolean+Operators+-28AND+OR-29.

**Reply by rida-elbahtouri** [reply]
Perfect that's what I was looking for, thank you so much @janko.
